specification/protocol/inference_rest.md

```diff
@@ -0,0 +1,367 @@
+| ------------- | ------------- | ------------- | ------------- | ------------- |
+| Inference | POST | v2/models/<model_name>[/versions/\<model_version\>]/infer | [$inference_request](#inference-request-json-object) | [$inference_response](#inference-response-json-object) |
+| Model Metadata | GET | v2/models/\<model_name\>[/versions/\<model_version\>] | | [$metadata_model_response](#model-metadata-response-json-object) | 
+| Server Ready | GET | v2/health/ready | | [$ready_server_response](#server-ready-response-json-object) | 
+| Server Live | GET | v2/health/live | | [$live_server_response](#server-live-response-json-objet)| 
+| Server Metadata | GET | v2 | | [$metadata_server_response](#server-metadata-response-json-object) |
+| Model Ready| GET   | v2/models/\<model_name\>[/versions/<model_version>]/ready |  | [$ready_model_response](#model-ready-response-json-object) |
+
+** path contents in `[]` are optional
+
+For more information regarding payload contents, see `Payload Contents`.
+
+The versions portion of the `Path` URLs (in `[]`) is shown as **optional** to allow implementations that don’t support versioning or for cases when the user does not want to specify a specific model version (in which case the server will choose a version based on its own policies).
+For example, if a model does not implement a version, the Model Metadata request path could look like `v2/model/my_model`. If the model has been configured to implement a version, the request path could look something like `v2/models/my_model/versions/v10`, where the version of the model is v10.
+
+<!-- // TODO: add example with -d inputs. -->
+
+### **API Definitions**
+
+| API  | Definition | 
+| --- | --- |
+| Inference | The `/infer` endpoint performs inference on a model. The response is the prediction result.| 
+| Model Metadata | The "model metadata" API is a per-model endpoint that returns details about the model passed in the path. | 
+| Server Ready | The “server ready” health API indicates if all the models are ready for inferencing. The “server ready” health API can be used directly to implement the Kubernetes readinessProbe |
+| Server Live | The “server live” health API indicates if the inference server is able to receive and respond to metadata and inference requests. The “server live” API can be used directly to implement the Kubernetes livenessProbe. |
+| Server Metadata | The "server metadata" API returns details describing the server. | 
+| Model Ready | The “model ready” health API indicates if a specific model is ready for inferencing. The model name and (optionally) version must be available in the URL. |
+
+### Health/Readiness/Liveness Probes
+
+The Model Readiness probe the question "Did the model download and is it able to serve requests?" and responds with the available model name(s). The Server Readiness/Liveness probes answer the question "Is my service and its infrastructure running, healthy, and able to receive and process requests?"
```

Comment:
Should this question be `"Was the model successfully downloaded and loaded into the server?"`. I thought it's the server that serves requests.

Reply:
good point, the model is loaded into the memory so the server is able to invoke the model's inference API upon receiving the request